docs/developer/devops/implementation.md

```diff
@@ -0,0 +1,104 @@
+the use of Gitbeaker as a wrapper, as this approach simplifies interactions
+with GitLab’s REST API and reduces the complexity of the project code.
+
+After testing the correctness of the API by implementing a `GitlabDriver`
+class, the APIs were integrated into the front-end. The latter was done by
+wiring up API endpoints to the front-end components, ensuring a seamless data
+flow. Unit and integration testing was done to ensure the coverage of all
+functional requirements and solve all problems regarding data consistency,
+performance, or user experience.
+
+## GitlabInstance
+
+The `GitlabInstance` class was created in order to manage the APIs and
+information related to the GitLab profile, the project, and the user-specific
+data stored in their account.
+
+The username and the token required to instantiate the Gitbeaker _Gitlab_
+component, which is required for making the API calls, are retrieved from the
+session storage, taking the _access_token_ of the user already logged into the
+DTaaS application.
+
+The initialisation of the `GitlabInstance` object is concluded with the
+execution of the `init()` method, which enables the retrieval and storage of
+the `projectId` and `triggerToken` attributes. The _projectId_ is a unique
+identifier for projects in GitLab and it is essential for subsequent API calls.
+For example, it is passed to the method that retrieves a _trigger token_,
+which is used to trigger CI/CD pipelines in GitLab.
+
+The objective of the `getDTSubfolders` method was to retrieve the names and
+corresponding descriptions of the DTs of the user, so that these could be shown
+at the front-end interface. This approach would obviate the user from having to
+input the name of a DT; hence, saving the user from possible error and
+inefficiencies arising from manual input. The user interface makes it easier
+for the user to deal with DTs by automatizing their selection and manages them
+more accurately. This implementation also eliminates the necessity for manual
+input from users for the access token and the username, which are automatically
+provided via the GitLab OAuth login.
+
+Furthermore, logs maintained in the `GitlabInstance` class improve awareness
+and transparency over the operations conducted. The final three methods are
+employed in conjunction to oversee the execution of a DT. In particular,
+individual logs are saved for each job in the pipeline, and the status of the
+latter is monitored so that, once the entire pipeline is complete, the results
+can be displayed in detail within the user interface. In this way, all statuses
+of each operation are logged for better debugging and performance analysis,
+including possible errors. Having trace logs exposed to the user means
+troubleshooting will be more effective and insight into execution and
+management of DTs will be gained, improving system reliability and user
+confidence.
+
+## DigitalTwin
+
```

Comment:
Please put a typescript code block containing class name and its methods. You can copy the class name and method signatures from codebase. The explanation makes sense w.r.t this diagram.

Pæease do this for all the classes explained in this PR.